File: src/main/java/com/codecool/repvault/application/DTOs/outgoing/ExerciseResponseDTO.java
package com.codecool.repvault.application.DTOs.outgoing;

import com.codecool.repvault.domain.entities.Exercise;
import com.codecool.repvault.domain.entities.ExerciseTemplate;
import com.codecool.repvault.domain.entities.enums.ExerciseEnum;
import lombok.Getter;

import java.util.List;

@Getter
public class ExerciseResponseDTO {
    private Long id;
    private ExerciseEnum type;
    private String description;
    private List<SetResponseDTO> sets;

    public ExerciseResponseDTO(Exercise exercise) {
        this.id = exercise.getId();
        this.type = exercise.getType();
        this.description = exercise.getDescription();
        this.sets = exercise.getSets().stream()
                .map(SetResponseDTO::new)
                .toList();
    }

    public ExerciseResponseDTO(ExerciseTemplate exerciseTemplate) {
        this.id = exerciseTemplate.getId();
        this.type = exerciseTemplate.getType();
        this.description = exerciseTemplate.getDescription();
        this.sets = exerciseTemplate.getSets().stream()
                .map(SetResponseDTO::new)
                .toList();
    }
}

File: src/main/java/com/codecool/repvault/application/DTOs/outgoing/ExerciseResponseDTO.kt
package com.codecool.repvault.application.DTOs.outgoing

import com.codecool.repvault.domain.entities.Exercise
import com.codecool.repvault.domain.entities.Set
import com.codecool.repvault.domain.entities.enums.ExerciseEnum

class ExerciseResponseDTO(exercise: Exercise) {
    val id: Long = exercise.id!!
    val type: ExerciseEnum = exercise.type!!
    val description: String? = exercise.description
    val sets: MutableList<SetResponseDTO> = exercise.sets.stream()
        .map<SetResponseDTO> { set: Set -> SetResponseDTO(set.id!!, set.type!!, set.reps, set.weight!!) }
        .toList()
}
